Automattic/jetpack
# Social: Move settings to a modal from the editor sidebar

This moves the Social Image Generator settings to a modal from the block editor sidebar. Eventually we should be able to move all the social settings for the post to a modal window.

<img width="694" alt="CleanShot 2023-07-19 at 14 55 21 png" src="https://github.com/Automattic/jetpack/assets/36671565/f37d5e4e-fe8b-4cbd-84eb-d852b2616df3">





Fixes #

## Proposed changes:

* Adds the new modal, where the Template picker and the custom header props live from now
* Saves on modal save

### Other information:


## Jetpack product discussion


1204930070946424-as-1204411047112864
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Open up the Editor with the Advanced plan, there should be only a Settings button now under the SIG panel
* Clicking it opens up the modal, where there is a preview on top, and settings below
* Play around with the settings, it should get reflected in the modal preview
  * Saving the modal will change the preview in the editor to the new one instantly
  * Saving the post and refreshing should result in seeing the most up to date version
